lib/jsdom/living/events/Event-impl.js

```diff
@@ -78,6 +80,57 @@ class EventImpl {
     }
   }
 
+  // https://dom.spec.whatwg.org/#dom-event-composedpath
+  composedPath() {
```

Comment:
The last failing tests are related to `composedPath` are still failing:
* [Event Path with a slot in a closed Shadow Root.](https://github.com/web-platform-tests/wpt/blob/master/shadow-dom/event-composed-path.html#L198-L219)
* [Event Path with slots in nested ShadowRoots: closed > closed.](https://github.com/web-platform-tests/wpt/blob/master/shadow-dom/event-composed-path.html#L251-L280)

I don't understand how the [composedPath algorithm](https://dom.spec.whatwg.org/#dom-event-composedpath) would work in the case of slot content in closed shadow trees.

----

Let's take for example: [Event Path with a slot in a closed Shadow Root.](https://github.com/web-platform-tests/wpt/blob/master/shadow-dom/event-composed-path.html#L198-L219)

```
test9
└── host1
    ├──#shadowRoot (closed)
    │   └──slot
    └── target
```

For an event dispatched on target, the following event path would get produced:

| item       | target | rootOfClosedTree | slotInSlosedTree |
|------------|--------|------------------|------------------|
| target     | target | false            | false            |
| slot       |        | false            | true             |
| shadowRoot |        | true             | false            |
| host1      |        | false            | false            |
| test9      |        | false            | false            |

Now if we consider target to the context object of composedPath the result would be: `[target, slot, shadowRoot, host1, test9]` while it should be `[target, host1, test9]`

| item       | hiddenSubtreeLevel | hasSeenCurrentTarget | reversedComposedPath                   |
|------------|--------------------|----------------------|----------------------------------------|
| target     | 0                  | true                 | test9, host1, shadowRoot, slot, target |
| slot       | 0                  | false                | test9, host1, shadowRoot, slot         |
| shadowRoot | 0                  | false                | test9, host1, shadowRoot               |
| host1      | 0                  | false                | test9, host1                           |
| test9      | 0                  | false                | test9                                  |